File: Codeforces Round 986 (Div. 2)/B_Alice_s_Adventures_in_Permuting.cpp
#include <iostream>
using namespace std;

using ll = long long;
int main()
{
    long long int n, b, c;
    int s;
    cin >> s;
    while (s--)
    {
        cin >> n >> b >> c;
        if (b == 0)
        {
            if (c >= n)
            {
                cout << n << endl;
                continue;
            }
            else if (c == n - 1 || c == n - 2)
            {
                cout << n - 1 << endl;
                continue;
            }
            else
            {
                cout << -1 << endl;
                continue;
            }
        }
        else
        {
            if (c >= n)
            {
                cout << n << endl;
                continue;
            }
            else
            {
                cout << n - max(0ll, 1 + (n - c - 1) / b) << endl;
                continue;
            }
        }
    }
    return 0;
}

File: Codeforces Round 986 (Div. 2)/B_Alice_s_Adventures_in_Permuting.py
for _ in range(int(input())):
    n, b, c = map(int, input().split())
    if b == 0:
        if c >= n:
            print(n)
        elif c in {n - 1, n - 2}:
            print(n - 1)
        else:
            print(-1)
    else:
        print(n if c >= n else n - max(0, 1 + (n - c - 1) // b))
